registry/proxy/proxyblobstore.go

```diff
@@ -50,6 +50,11 @@ func (pbs *proxyBlobStore) copyContent(ctx context.Context, dgst digest.Digest,
 		setResponseHeaders(w, desc.Size, desc.MediaType, dgst)
 	}
 
+        if ! copydata{
+                // if HEAD then skip copy data
+                return distribution.Descriptor{},nil
```

Comment:
Looks like the head case ignore the returned descriptor but this function should still return `desc` in this case